core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1239,6 +1239,9 @@ private[spark] class DAGScheduler(
           markMapStageJobsAsFinished(stage)
         case stage : ResultStage =>
           logDebug(s"Stage ${stage} is actually done; (partitions: ${stage.numPartitions})")
```

Comment:
would this happen for normal jobs?

Reply:
This condition is guarded by a check when ``ResultStage`` itself has zero tasks to run. So it would be skipped for normal jobs

Reply:
I mean, does `RDD#count` have the same bug when rdd is empty? I think it doesn't but I'm trying to understand why it's an issue only for approximate jobs.

Reply:
I think this is related to SPARK-26714 / https://github.com/apache/spark/pull/23637 -- maybe the same change from `DAGScheduler.submitJob` is needed in `DAGScheduler.runApproximateJob`

Reply:
I think @squito suggestion is better, i have changed my PR as per it. Please review